### PR TITLE
Sprout Utility/UtilityConnection for connecting a Space to an external Service

### PR DESCRIPTION
**Pull request by zspencer**
This is very rough, but the goal here is to make it so the Plaid (or possibly DocuSign) configuration can be set on a per-space basis for #186; as well as open up opportunities for folks to connect their existing infrastructure for furniture that leverages it.

### CODE REVIEW

File: convene-web/app/lib/utilities.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Utilities are External Services that provide functionality to a piece of
```
Comment (anaulin) [inline convene-web/app/lib/utilities.rb:1]: This comment immediately makes me think "Should we call this an `ExternalService` then?". Naming is hard, and I am myself a fan of using "Utility" when I can think of anything better, but I also recognize it is kinda like calling something "Thingie" -- not a huge value add. 😹

File: convene-web/app/lib/utilities.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+# - A Jitsi-Meet or BigBlueButton instance for videoconferencing
+# - An Airtable Base or Google Apps account that acts as a human-editable
+#   backend-data store for a piece of Furniture.
+# - A Dropbox, OneCloud or Google Drive account for file sharing.
+# - ???
+#
+module Utilities
+  # @return [Utility] configured as appropriate given the connection
+  def self.from_connection(connection)
+    Utility.new(settings: connection)
```
Comment (anaulin) [inline convene-web/app/lib/utilities.rb:17]: Sounds like the "connection" is actually a "configuration" or "settings" (as opposed to a server-to-server connection, which is what I would have assumed from the name).

File: convene-web/app/models/utility_connection.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
+# Links a {Utilities::Utility} to a {Space} or {Person}.
+class UtilityConnection < ApplicationRecord
```
Comment (anaulin) [inline convene-web/app/models/utility_connection.rb:2]: Ok, this is silly, but hear me out: if it is a "utility" in the gas, water, and electricity sense, then this is class is really a "hook up". (I told you this was silly.)

File: convene-web/app/models/utility_connection.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  belongs_to :space
+
+  # Human-friendly name for disambiguating when a single kind of Utility has
+  # multiple Connections.
+  attribute :name
+  attribute :description
+
+  # Which type of {Utility} is connected, for example Airtable or  Asana or
+  # Jitsi-Meet or BBB or Plaid or Stripe.
+  attribute :utility_type
```
Comment (anaulin) [inline convene-web/app/models/utility_connection.rb:13]: Noodling out loud: would this eventually be an enum with a registry?

**Review by anaulin — COMMENTED** [review]
This structure makes sense to me so far. I left some thinking-out-loud comments that are around naming, trying to make things a little bit more self-explanatory to a future reader of the code.

File: convene-web/app/models/utility_connection.rb (hunk 3 above)
Comment (zspencer) [reply convene-web/app/models/utility_connection.rb:2]: No that's amazing I love it.

File: convene-web/app/models/utility_connection.rb (hunk 4 above)
Comment (zspencer) [reply convene-web/app/models/utility_connection.rb:13]: Yea, I wonder if there's a Registry class provided by Rails  that we could use for this...

File: convene-web/app/lib/utilities.rb (hunk 2 above)
Comment (zspencer) [reply convene-web/app/lib/utilities.rb:17]: Yea, that's a good point

File: convene-web/app/models/utility_connection.rb (hunk 4 above)
Comment (zspencer) [reply convene-web/app/models/utility_connection.rb]: There is not a registry class provide by Rails for this :(

**Pull request merged**